### PR TITLE
EVAKA-4244: fix and improve duplicate people merge tool

### PR DESCRIPTION
**Pull request by Joosakur**

### CODE REVIEW

File: service/src/integrationTest/kotlin/fi/espoo/evaka/pis/PersonIntegrationTest.kt [hunk 1]
```diff
@@ -62,6 +62,61 @@ class PersonIntegrationTest : PureJdbiTest() {
...
+                PersonReference("invoice_row", "child"),
+                PersonReference("koski_study_right", "child_id"),
+                PersonReference("messaging_blocklist", "blocked_recipient"),
+                PersonReference("messaging_blocklist", "child_id"),
+                PersonReference("new_fee_decision", "head_of_family_id"),
+                PersonReference("new_fee_decision", "partner_id"),
+                PersonReference("new_fee_decision_child", "child_id"),
+                PersonReference("placement", "child_id"),
+                PersonReference("service_need", "child_id"),
+                PersonReference("varda_child", "person_id"),
```
Comment (Joosakur) [inline service/src/integrationTest/kotlin/fi/espoo/evaka/pis/PersonIntegrationTest.kt:111]: should varda_child data be transferred or deleted?

File: service/src/integrationTest/kotlin/fi/espoo/evaka/pis/PersonIntegrationTest.kt [hunk 2]
```diff
@@ -62,6 +62,61 @@ class PersonIntegrationTest : PureJdbiTest() {
...
+                PersonReference("fee_decision", "head_of_family"),
+                PersonReference("fee_decision", "partner"),
+                PersonReference("fee_decision_part", "child"),
+                PersonReference("fridge_child", "child_id"),
+                PersonReference("fridge_child", "head_of_child"),
+                PersonReference("fridge_partner", "person_id"),
+                PersonReference("income", "person_id"),
+                PersonReference("invoice", "head_of_family"),
+                PersonReference("invoice_row", "child"),
+                PersonReference("koski_study_right", "child_id"),
```
Comment (Joosakur) [inline service/src/integrationTest/kotlin/fi/espoo/evaka/pis/PersonIntegrationTest.kt:103]: should koski_study_right data be transferred or deleted?

File: service/src/integrationTest/kotlin/fi/espoo/evaka/pis/PersonIntegrationTest.kt (hunk 1 above)
Comment (jaahon) [reply service/src/integrationTest/kotlin/fi/espoo/evaka/pis/PersonIntegrationTest.kt:111]: Probably just transferred.

File: service/src/main/kotlin/fi/espoo/evaka/pis/service/MergeService.kt [hunk 3]
```diff
@@ -106,18 +96,33 @@ class MergeService(private val asyncJobRunner: AsyncJobRunner) {
...
+                """.trimIndent()
+            }} +
+                (SELECT count(*) FROM message WHERE sender_id = (SELECT id FROM message_account WHERE person_id = :id)) +
+                (SELECT count(*) FROM message_content WHERE author_id = (SELECT id FROM message_account WHERE person_id = :id)) +
+                (SELECT count(*) FROM message_recipients WHERE recipient_id = (SELECT id FROM message_account WHERE person_id = :id)) +
+                (SELECT count(*) FROM message_draft WHERE account_id = (SELECT id FROM message_account WHERE person_id = :id)) AS count;
             """.trimIndent()
 
-        tx.createUpdate(sql1).bind("id", id).execute()
+        val referenceCount = tx.createQuery(sql1).bind("id", id).mapTo<Int>().first()
```
Comment (akheron) [inline service/src/main/kotlin/fi/espoo/evaka/pis/service/MergeService.kt:116]:
```suggestion
        val referenceCount = tx.createQuery(sql1).bind("id", id).mapTo<Int>().one()
```

**Review by akheron — APPROVED** [review]

**Pull request merged**